lib/benchee/formatters/console.ex

```diff
@@ -145,11 +155,65 @@ defmodule Benchee.Formatters.Console do
     units = Conversion.units(sorted_scenarios, scaling_strategy)
     label_width = label_width(sorted_scenarios)
 
-    [column_descriptors(label_width) |
+    output = [column_descriptors(label_width) |
       scenario_reports(sorted_scenarios, units, label_width)
       ++ comparison_report(sorted_scenarios, units, label_width, config)]
+
+    if config.extended_options do
+      output ++ [descriptor("Extended options") |
+        [extended_column_descriptors(label_width)]
+          ++ extended_options_reports(sorted_scenarios, units, label_width)]
+    else
+      output
+    end
+  end
+
+  @spec extended_options_reports([Scenario.t], map, integer) :: [any, ...]
+  defp extended_options_reports(scenarios, units, label_width) do
+    Enum.map(scenarios, fn(scenario) ->
+              format_scenario_extended(scenario, units, label_width) end)
+  end
+
+  @spec format_scenario_extended(Scenario.t, map, integer) :: String.t
```

Comment:
This is where the dialyzer error comes from... typing looks right to me and some builds also passed. Sometimes... dialyzer... 

```
lib/benchee/formatters/console.ex:177: Invalid type specification for function 'Elixir.Benchee.Formatters.Console':format_scenario_extended/3. The success typing is (#{'__struct__':='Elixir.Benchee.Benchmark.Scenario', 'job_name':=_, 'run_time_statistics':=#{'__struct__':='Elixir.Benchee.Statistics', 'maximum':=number(), 'minimum':=number(), 'mode':=[number()], 'sample_size':=_, _=>_}, _=>_},#{'ips':=#{'__struct__':='Elixir.Benchee.Conversion.Unit', 'label':=binary(), 'long':=binary(), 'magnitude':=non_neg_integer(), 'name':=atom()}, 'run_time':=#{'__struct__':='Elixir.Benchee.Conversion.Unit', 'label':=binary(), 'long':=binary(), 'magnitude':=non_neg_integer(), 'name':=atom()}},integer()) -> binary()
```

Reply:
So this is failing locally for me, but I cannot for the life of me discern why this is failing. This method here even uses the exact same typespec and does not fail... https://github.com/lwalter/benchee/blob/8d87ccf0c4cc4354781714370f6c80a13f3391b3/lib/benchee/formatters/console.ex#L186